numpy/core/_internal.py

```diff
@@ -263,6 +273,14 @@ def __init__(self, array, ptr=None):
         else:
             self._zerod = False
 
+    def __getattribute__(self, name):
```

Comment:
As opposed to `__getattribute__`, `__getattr__` is not called always, which might be nice (https://docs.python.org/3/reference/datamodel.html#object.__getattr__)?

Although, I think making them properties that give the warning is probably the most elegant.  If it doesn't seem nicer to you, also fine.  I like deprecting these :).

Reply:
Done as of https://github.com/numpy/numpy/pull/19031/commits/d74ebdd71ba2b38b3f9b309ab416a35bc5f7bd7b.